templates/aws-vpc.template.yaml

```diff
@@ -303,849 +345,911 @@ Parameters:
     Default: default
     Description: The allowed tenancy of instances launched into the VPC
     Type: String
+Rules:
+  NAT:
+    RuleCondition: !Equals [!Ref CreateNATGateways, "true"]
+    Assertions:
+      - Assert: !And
+          - !Equals [!Ref CreatePrivateSubnets, "true"]
+          - !Equals [!Ref CreatePublicSubnets, "true"]
+        AssertDescription: To enable NAT gateways you must have both CreatePrivateSubnets and CreatePublicSubnets set to 'true'
+  Subnets:
+    Assertions:
+      - Assert: !Or
+          - !Equals [!Ref CreatePrivateSubnets, "true"]
+          - !Equals [!Ref CreatePublicSubnets, "true"]
+        AssertDescription: At least one of CreatePublicSubnets or CreatePrivateSubnets must be set to 'true'
 Conditions:
   3AZCondition: !Or
     - !Equals
-      - !Ref 'NumberOfAZs'
-      - '3'
-    - !Condition '4AZCondition'
+      - !Ref "NumberOfAZs"
+      - "3"
+    - !Condition "4AZCondition"
   4AZCondition: !Equals
-    - !Ref 'NumberOfAZs'
-    - '4'
+    - !Ref "NumberOfAZs"
+    - "4"
   AdditionalPrivateSubnetsCondition: !And
     - !Equals
-      - !Ref 'CreatePrivateSubnets'
-      - 'true'
+      - !Ref "CreatePrivateSubnets"
+      - "true"
     - !Equals
-      - !Ref 'CreateAdditionalPrivateSubnets'
-      - 'true'
+      - !Ref "CreateAdditionalPrivateSubnets"
+      - "true"
   AdditionalPrivateSubnets&3AZCondition: !And
-    - !Condition 'AdditionalPrivateSubnetsCondition'
-    - !Condition '3AZCondition'
+    - !Condition "AdditionalPrivateSubnetsCondition"
+    - !Condition "3AZCondition"
   AdditionalPrivateSubnets&4AZCondition: !And
-    - !Condition 'AdditionalPrivateSubnetsCondition'
-    - !Condition '4AZCondition'
+    - !Condition "AdditionalPrivateSubnetsCondition"
+    - !Condition "4AZCondition"
+
+  AdditionalPrivateSubnets&PublicSubnets&NatGatewaysCondition: !And
+    - !Condition "AdditionalPrivateSubnetsCondition"
+    - !Condition "PublicSubnetsCondition"
+    - !Condition "NATGatewaysCondition"
+  AdditionalPrivateSubnets&PublicSubnets&NatGateways&3AZCondition: !And
+    - !Condition "AdditionalPrivateSubnets&3AZCondition"
+    - !Condition "PublicSubnetsCondition"
+    - !Condition "NATGatewaysCondition"
+  AdditionalPrivateSubnets&PublicSubnets&NatGateways&4AZCondition: !And
+    - !Condition "AdditionalPrivateSubnets&4AZCondition"
+    - !Condition "PublicSubnetsCondition"
+    - !Condition "NATGatewaysCondition"
+
   GovCloudCondition: !Equals
-    - !Ref 'AWS::Region'
+    - !Ref "AWS::Region"
```

Comment:
This condition is no longer needed, unless I'm missing something.

Reply:
We agreed to limit the changes to only the items related to privatizing the VPC